client/lib/desktop/index.js

```diff
@@ -121,77 +120,20 @@ const Desktop = {
 		ipc.send( 'unread-notices-count', unseenCount );
 	},
 
-	onNewNotification: function ( event ) {
-		const noteWithMeta = event.detail;
-		const { note } = noteWithMeta;
-		debug( `Received notification: ${ note.id }` );
-		const siteTitle = getSiteTitle( this.store.getState(), note.meta.ids.site );
-		ipc.send( `received-notification`, {
-			siteTitle,
-			...noteWithMeta,
-		} );
-	},
-
-	onNotificationClicked: function ( _, noteWithMeta ) {
-		const { note, isApproved } = noteWithMeta;
-		debug( `Notification ${ note.id } clicked` );
+	onNotificationClicked: function ( _, notification ) {
+		debug( `Notification ${ notification.id } clicked` );
 
-		const noteId = note.id;
-		const linkType = note.type;
-		const siteId = note.meta.ids.site;
-		const postId = note.meta.ids.post;
-		const commentId = note.meta.ids.comment;
-		let fallBackToNotificationsPanel = false;
+		const { id, type } = notification;
 
 		// TODO: Make this a desktop-specific Tracks event.
 		this.store.dispatch( recordTracksEventAction( 'calypso_web_push_notification_clicked' ), {
-			push_notification_note_id: noteId,
-			push_notification_type: linkType,
+			push_notification_note_id: id,
+			push_notification_type: type,
 		} );
+	},
 
-		// Tell the notifications panel to mark this note as read.
-		window.dispatchEvent(
-			new window.CustomEvent( 'desktop-notification-mark-as-read', {
-				detail: {
-					noteId,
-				},
-			} )
-		);
-
-		switch ( linkType ) {
-			case 'post':
-				this.navigate( `/read/blogs/${ siteId }/posts/${ postId }` );
-				break;
-			case 'comment':
-				{
-					// If the note is approved, navigate to the comment URL within Calypso.
-					// Otherwise open and display Calypso's notifications panel.
-					if ( isApproved ) {
-						this.navigate( `/read/blogs/${ siteId }/posts/${ postId }#comment-${ commentId }` );
-					} else {
-						fallBackToNotificationsPanel = true;
-					}
-				}
-				break;
-			case 'comment_like':
-				this.navigate( `/read/blogs/${ siteId }/posts/${ postId }#comment-${ commentId }` );
-				break;
-			case 'site':
-				this.navigate( `/read/blogs/${ siteId }` );
-				break;
-			default:
-				fallBackToNotificationsPanel = true;
-		}
-
-		// Fall back to notifications panel for unhandled note types.
-		if ( fallBackToNotificationsPanel ) {
-			setTimeout( () => {
-				this.navigate( '/' );
-				if ( ! isNotificationsOpen( this.store.getState() ) ) {
-					this.toggleNotificationsPanel();
-				}
-			}, 1000 );
-		}
+	onNotificationsPanelRefresh: function () {
+		this.store.dispatch( forceNotificationsRefresh( true ) );
```

Comment:
I think this should simply import the `refreshNotes` method from the `notifications` module and call it. Adding the Redux action and reducer framework around it should be completely unnecessary.

Reply:
I agree this seems a little convoluted but I can't find a prior example of the `client` code importing helpers from `apps/` directory. I gave this approach a try in 497823d3b9b119fbaa29379d61a1974f24f5e476 but ultimately reverted due to a couple issues: 

- for some reason, it didn't work to refresh the notifications panel (confirmed by logging activity from the notifications `rest-client`). Not sure why. 
- the linter also complained about the import in `client/lib/desktop/index.js`:

```
import { refreshNotes } from '../../../apps/notifications/src/panel/notifications';
// error: file is not under 'rootDir'. `rootDir` is expected to contain all source files.
```

I can't find a prior example of `client` code importing something from the `apps` directory, but happy to revisit this as a refactor, though.